### ISSUE TITLE
Mixin documentation doesn't work

### ISSUE THREAD

**Issue by TrapII** [jsdoc2md/jsdoc-to-markdown#146]
If I generate doc from this simple mixin :1234: 

```js
/**
 * Create an serialalizable class.
 * @mixin SerializeMixin
 * @param {*} superclass - The class to mix onto.
 * @return The mixin class.
 * @mixin SerialMixin
 */
const SerialMixin = (superclass) => class extends superclass {
  
  /**
   * Adds a string to the stream
   * @param {string} str - The string to append to the stream
   * @memberof SerialMixin#
   */
  addString(str) {
  }
}
```

This produce a result where the method names are prefixed with **undefined** : `serialMixin.undefinedaddString(str).`
I tried various combination (with lendof, ...) but the bug seems to be there whatever I tried.
The version of the jsdoc-to-markdown is 3.0.2. I also uses the dmd-bitbucket plugin in version 0.1.10.

There is a corresponding issue to this one in the jsdoc repository : [here](https://github.com/jsdoc3/jsdoc/issues/1409). But seems corrected.

**Comment by 75lb** [comment]
Could you confirm the code you pasted documents as expected using jsdoc directly, please?

**Comment by lemnis** [comment]
Bumped into the same issue, the JSON of jsdoc reports as longname `SerialMixin#addString` and jsdoc2md reports `SerialMixin#undefinedaddString`.

Tested with jsdoc@3.5.5 and jsdoc-to-markdown@3.0.3

**Comment by 75lb** [comment]
This is a jsdoc issue.. if you save the original post sample code into a file called `issue-146.js` and run this: 

```
$ jsdoc -X issue-146.js
```

The output (below) contains `undefinedaddString`, which in turn appears in jsdoc2md output. 

```
[
    {
        "comment": "/**\n * Create an serialalizable class.\n * @mixin SerializeMixin\n * @param {*} superclass - The class to mix onto.\n * @return The mixin class.\n * @mixin SerialMixin\n */",
        "meta": {
            "range": [
                0,
                167
            ],
            "filename": "issue-146.js",
            "lineno": 1,
            "path": "/Users/lloyd/Documents/jsdoc2md/testbed/tmp",
            "code": {}
        },
        "description": "Create an serialalizable class.",
        "kind": "mixin",
        "name": "SerialMixin",
        "params": [
            {
                "type": {
                    "names": [
                        "*"
                    ]
                },
                "description": "The class to mix onto.",
                "name": "superclass"
            }
        ],
        "returns": [
            {
                "description": "The mixin class."
            }
        ],
        "longname": "SerialMixin",
        "scope": "global"
    },
    {
        "comment": "",
        "meta": {
            "range": [
                174,
                391
            ],
            "filename": "issue-146.js",
            "lineno": 8,
            "path": "/Users/lloyd/Documents/jsdoc2md/testbed/tmp",
            "code": {
                "id": "astnode100000002",
                "name": "SerialMixin",
                "type": "ArrowFunctionExpression"
            },
            "vars": {
                "undefined": null
            }
        },
        "undocumented": true,
        "name": "SerialMixin",
        "longname": "SerialMixin",
        "kind": "function",
        "scope": "global"
    },
    {
        "comment": "/**\n   * Adds a string to the stream\n   * @param {string} str - The string to append to the stream\n   * @memberof SerialMixin#\n   */",
        "meta": {
            "range": [
                369,
                389
            ],
            "filename": "issue-146.js",
            "lineno": 15,
            "path": "/Users/lloyd/Documents/jsdoc2md/testbed/tmp",
            "code": {
                "id": "astnode100000009",
                "name": "undefinedaddString",
                "type": "MethodDefinition",
                "paramnames": [
                    "str"
                ]
            },
            "vars": {
                "": null
            }
        },
        "description": "Adds a string to the stream",
        "params": [
            {
                "type": {
                    "names": [
                        "string"
                    ]
                },
                "description": "The string to append to the stream",
                "name": "str"
            }
        ],
        "memberof": "SerialMixin",
        "name": "undefinedaddString",
        "longname": "SerialMixin#undefinedaddString",
        "kind": "function",
        "scope": "instance"
    },
    {
        "kind": "package",
        "longname": "package:undefined",
        "files": [
            "/Users/lloyd/Documents/jsdoc2md/testbed/tmp/issue-146.js"
        ]
    }
]
```

Please post an issue with the [jsdoc](https://github.com/jsdoc3/jsdoc) project and link back to this issue. 👍

**Comment by lemnis** [comment]
Which version are you at? Its an issue that is fixed in 3.5.4. It's an issue with the (out of date) fork...

**Comment by 75lb** [comment]
Fixed and released in [v4.0.0](https://github.com/jsdoc2md/jsdoc-to-markdown/releases/tag/v4.0.0) 👍
